$NetBSD$

Adjust Makefile to work well for pkgsrc, specific path, options for compilation
 and link

--- Makefile.orig	2016-04-06 03:11:44.000000000 +0000
+++ Makefile
@@ -2,37 +2,32 @@
 
 LUA_VER         ?= 5.1
 LUA_DIR         ?= /usr
-LUA_LIBDIR      := $(LUA_DIR)/lib/lua/$(LUA_VER)/term
-LUA_INC         := $(LUA_DIR)/include/lua$(LUA_VER)
-LUA_SHARE       := $(LUA_DIR)/share/lua/$(LUA_VER)/term
+LUA_LIBDIR      ?= $(LUA_DIR)/lib/lua/$(LUA_VER)/term
+LUA_INC         ?= $(LUA_DIR)/include/lua$(LUA_VER)
+LUA_SHARE       ?= $(LUA_DIR)/share/lua/$(LUA_VER)/term
 CWARNS          := -Wall -pedantic
 CFLAGS          += $(CWARNS) -O3 -I$(LUA_INC) -fPIC
 LIB_OPTION      := -shared
 
 SONAME          := core.so
-SONAMEV         := $(SONAME).1
-LIBRARY         := $(SONAMEV).0.1
+LIBRARY         := $(SONAME)
 SRC             := core.c
 OBJ             := $(patsubst %.c, %.o, $(SRC))
 
-FILES           := term/init.lua term/cursor.lua term/colors.lua
-
-all: $(LIBRARY) $(SONAMEV) $(SONAME)
+EXTRA_LDFLAGS_Darwin	= -install_name $(LUA_LIBDIR)/$(SONAME) -undefined dynamic_lookup
 
-$(SONAMEV):
-	ln -s $(LIBRARY) $@
+FILES           := term/init.lua term/cursor.lua term/colors.lua
 
-$(SONAME):
-	ln -s $(SONAMEV) $@
+all: $(LIBRARY)
 
 $(LIBRARY): $(OBJ)
-	$(CC) $(CFLAGS) $(LIB_OPTION) -o $(LIBRARY) $(OBJ) -lc
+	$(CC) $(LIB_OPTION) $(EXTRA_LDFLAGS_$(OPSYS)) -o $(LIBRARY) $(OBJ) -lc
 
 install:
-	mkdir -p $(LUA_LIBDIR)
-	cp $(SONAME) $(LUA_LIBDIR)
-	mkdir -p $(LUA_SHARE)
-	cp $(FILES) $(LUA_SHARE)
+	mkdir -p $(DESTDIR)$(LUA_LIBDIR)
+	cp $(SONAME) $(DESTDIR)$(LUA_LIBDIR)
+	mkdir -p $(DESTDIR)$(LUA_SHARE)
+	cp $(FILES) $(DESTDIR)$(LUA_SHARE)
 
 clean:
-	$(RM) $(LIBRARY) $(SONAMEV) $(SONAME) *.o
+	$(RM) $(LIBRARY) $(SONAME) *.o
